api/auth/controllers.py

```diff
@@ -54,70 +37,69 @@ def create_user(self):
         email = data.get('email')
         password = data.get('password')
         phone_number = data.get('phone_number')
-
-        user = User(firstname, lastname, othernames, username, email, password, phone_number)
-#         # validate user
-#         error = user.validate_user_input()
-#         base_error = user.validate_base_input()
-#         if error:
-#             return jsonify({
-#                 "status": 400,
-#                 "error": error
-#             }), 400
-#         if base_error:
-#             return jsonify({
-#                 "status": 400,
-#                 "error": base_error
-#             }), 400
-#         # check if user exists
-#         user_exists = users.find_user_by_username(username)
-#         if user_exists:
-#             return jsonify({
-#                 "status": 202,
-#                 "message": "User already exists. Please login."
-#             }), 202
-        db.insert_user_data(firstname, lastname, othernames, username, email, password, phone_number)
-
+        created_on = datetime.datetime.now()
+        # validate user input
+        validate_input = user_validations.validate_user_input(username, firstname, lastname,
+                                                              othernames, phone_number)
+        validate_email = user_validations.validate_user_email(email)
+        validate_password = user_validations.validate_user_password(password)
+        if validate_input:
+            return jsonify({
+                "status": 400,
+                "error": validate_input
+            }), 400
+        if validate_email:
+            return jsonify({
+                "status": 400,
+                "error": validate_email
+            }), 400
+        if validate_password:
+            return jsonify({
+                "status": 400,
+                "error": validate_password
+            }), 400
+        # check if user exists
+        user_exists = db.get_by_argument('users', 'username', username)
+        if user_exists:
+            return jsonify({
+                "status": 202,
```

Comment:
In case of duplicate data, a 409 status code would be more meaningful. More info on this [here](https://www.mnot.net/blog/2017/05/11/status_codes)

Reply:
Working on it